holidays/countries/gibraltar.py

```diff
@@ -0,0 +1,220 @@
+            else tr("Queen's Birthday")
+        )
+        sovereign_birthday_dts = {
+            2000: (JUN, 19),
+            2001: (JUN, 18),
+            2006: (JUN, 19),
+            2007: (JUN, 18),
+            2012: (JUN, 18),
+            2013: (JUN, 17),
+            2017: (JUN, 19),
+            2019: (JUN, 17),
+        }
+        if dt := sovereign_birthday_dts.get(self._year):
+            self._add_holiday(name, dt)
+        elif self._year <= 2022:
+            self._add_holiday_2_days_past_2nd_sat_of_jun(name)
+        else:
+            self._add_holiday_3rd_mon_of_jun(name)
+
+        name = (
+            # Summer Bank Holiday.
+            tr("Summer Bank Holiday")
+            if 2002 <= self._year <= 2007
+            # Late Summer Bank Holiday.
+            else tr("Late Summer Bank Holiday")
+        )
+        self._add_holiday_last_mon_of_aug(name)
+
+        # Gibraltar National Day.
+        name = tr("Gibraltar National Day")
+        national_day_dts = {
+            2016: (SEP, 5),
+            2017: (SEP, 4),
+        }
+        if dt := national_day_dts.get(self._year):
+            self._add_holiday(name, dt)
+        else:
+            self._add_observed(self._add_holiday_sep_10(name))
+
+        self._add_observed(
+            # Christmas Day.
+            self._add_christmas_day(tr("Christmas Day")),
+            rule=SAT_SUN_TO_NEXT_MON_TUE,
+        )
+
+        self._add_observed(
+            # Boxing Day.
+            self._add_christmas_day_two(tr("Boxing Day")),
+            rule=SAT_SUN_TO_NEXT_MON_TUE,
+        )
+
+
+class GI(Gibraltar):
+    pass
+
+
+class GIB(Gibraltar):
+    pass
+
+
+class GibraltarStaticHolidays(StaticHolidays):
+    """Gibraltar Special Holidays.
+
+    References:
+        * [Tercentenary Holiday](https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-2003-208)
+        * [12th January, 2009](https://web.archive.org/web/20250507113355/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-2008-2215/version/25-09-2008)
+        * [Queen's Diamond Jubilee](https://web.archive.org/web/20250513201915/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-no-2-order-2011-2988)
+        * [Evacuation Commemoration Day, 2015](https://web.archive.org/web/20250713063810/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-no-2-order-2015-3921/download)
+        * [75th Anniversary of VE Day, 2020](https://web.archive.org/web/20250426144346/https://www.gibraltar.gov.gi/press-releases/bank-and-public-holidays-2020-5363)
+        * [Platinum Jubilee](https://web.archive.org/web/20250508175822/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-2021-6286/version/21-10-2021)
+        * [Special King's Coronation Bank Holiday, 2023](https://web.archive.org/web/20250508180948/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-2022-6735/version/22-02-2023)
```

Comment:
```suggestion
    """Gibraltar special holidays.

    References:
        * [Evacuation Commemoration Day, 2015](https://web.archive.org/web/20250714174537/https://www.gibraltarlaws.gov.gi/uploads/legislations/banking-and-financial-dealings/B&P%20Holidays/2015s112.pdf)
```
Rest of the links are duplicated from main class references.